provider/k8s.go

```diff
@@ -633,6 +637,37 @@ func (store genericFileStore) ServeDirectory(dir string) (Iterator, error) {
 	return parquetIteratorOverMultipleFiles(fileParts, store)
 }
 
+func (store genericFileStore) Upload(sourcePath string, destPath string) error {
```

Comment:
Since the filestore already takes a container path as an argument, does it make sense to chance `destPath` to `filename` or allow it to be a subpath?

Reply:
I think the subpath makes sense because you can store it in different locations especially if later on, we support manual backups.